lib/api/model/model.dart

```diff
@@ -315,6 +315,11 @@ enum UserRole{
   final int? apiValue;
 
   int? toJson() => apiValue;
+
+  bool isAtLeast(UserRole threshold) {
+    // Roles with more privilege have lower [apiValue].
+    return (apiValue ?? 0) <= (threshold.apiValue ?? 0);
```

Comment:
Looking at this, I was thinking about why `0` is chosen as the default value for both operands. Considering the context this helper is used, I suppose that we want to fallback to the behavior that, when the role has an unknown value, we allow the user to post.

This consideration is specific to the posting permission feature we are implementing here. I think the best thing to do is to assert that both api values are non-null; then we handle the case when either of them is null in `hasPostingPermission`.

Reply:
Right. If we're going to allow the user to post when `threshold.apiValue == null`, then it should be:
```dart
return (apiValue ?? 0) <= (threshold.apiValue ?? double.maxFinite);
```

Also, as Chris mentioned in [#886 comment](https://github.com/zulip/zulip-flutter/pull/886#discussion_r1796280370), we're not going to expect any new values thus none of these will become null.